### PR TITLE
API 3rd party transaction signer integration

### PR DESCRIPTION
**Pull request by mickvandijke**
The goal of this PR is to allow 3rd party wallet apps to sign (and send) transactions that would otherwise be automatically signed and send within the API with the private key of a wallet.

3rd party wallet apps do not expose the private key of a wallet, so we can't pass that to the API. Alternative API endpoints have to be provided that allow us to skip automatic data payments and submit proof of payments directly instead.

The transaction input data will still be formed in the API however, to make the signing and sending of transactions from the 3rd party wallet side as easy as possible.

### CODE REVIEW

File: evmlib/src/external_signer.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Network token contract error: {0}")]
+    NetworkTokenContract(#[from] network_token::Error),
+    #[error("Chunk payments contract error: {0}")]
+    ChunkPaymentsContract(#[from] data_payments::error::Error),
+}
+
+#[allow(dead_code)]
```
Comment (b-zee) [inline evmlib/src/external_signer.rs]: Why are these needed for `pub`?

File: evmlib/src/external_signer.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    pub to: Address,
+    pub approve_spender: Address,
+    pub approve_amount: Amount,
+}
+
+#[allow(dead_code)]
+/// Use this wallet to pay for chunks in batched transfer transactions.
+/// If the amount of transfers is more than one transaction can contain, the transfers will be split up over multiple transactions.
+///
+/// Returns PayForQuotesCalldataReturnType, containing calldata of the transaction batches along with the approval details for the spender.
```
Comment (b-zee) [inline evmlib/src/external_signer.rs:61]: What exactly is 'call data'?

File: evmlib/src/external_signer.rs (hunk 1 above)
Comment (mickvandijke) [reply evmlib/src/external_signer.rs]: The Error struct? It needs the same visibility as the `pay_for_quotes_calldata` function that potentially returns it

File: evmlib/src/external_signer.rs (hunk 2 above)
Comment (mickvandijke) [reply evmlib/src/external_signer.rs:61]: I have used calldata as a general term to refer to the data that is needed to form a transaction, including transaction input and the to address. `Calldata` the type is an alias to `Bytes` and should be set as the input of a transaction.

Example: https://github.com/mickvandijke/safe_network/blob/metamask-integration/autonomi/tests/external_signer.rs#L45

File: evmlib/src/external_signer.rs (hunk 1 above)
Comment (b-zee) [reply evmlib/src/external_signer.rs]: I meant all `#[allow(dead_code)]` attributes in those file, sorry, don't know why GitHub showed the lines before that.

**Review by b-zee — APPROVED** [review]

**Review by grumbach — APPROVED** [review]
Amazing work!

File: evmlib/src/external_signer.rs (hunk 1 above)
Comment (mickvandijke) [reply evmlib/src/external_signer.rs]: Good catch, removed them all now
